streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoinTest.java

```diff
@@ -1382,7 +1394,7 @@ public void testAsymmetricWindowingAfter() {
         joined = stream1.join(
             stream2,
             MockValueJoiner.TOSTRING_JOINER,
-            JoinWindows.ofTimeDifferenceWithNoGrace(ofMillis(0)).after(ofMillis(100)),
+            JoinWindows.ofTimeDifferenceAndGrace(ofMillis(0), ofMillis(4)).after(ofMillis(100)),
```

Comment:
As above: this test is an inner-join and setup to never drop record due to grace -- we should set a larger value and also check "dropped record sensor".

It would also make sense to add the same extension to send additional "late" data at the end of this test?